### PR TITLE
added angular dependency for npm and bower

### LINKED ISSUE #275: Declare dependencies in package.json and bower.json
#### Expected behavior

- When installing from bower/npm, the dependencies (Angular 1.3+) should be installed (if not already)

#### Actual behavior

- Does not install Angular when installing from bower/npm

#### Steps to reproduce the behavior

Install from bower or npm in a new empty project, Angular 1.3 will not be installed.

I thought I had addressed this in #80, but apparently all I did was publish to npm. 

Need to also add the following to package.json and bower.json.

```
"dependencies": {
  "angular": "^1.3.0"
}
```

If we're interested in supporting browserify (no one has requested this so far) we'd also need to add an index.js [as described in this article](http://blog.npmjs.org/post/114584444410/using-angulars-new-improved-browserify-support). That could be it's own issue, b/c we'd also want to create a test page to verify that works.

### PR DESCRIPTION
**Pull request by jwasilgeo**
Proposed changes:
  - Should be able to automatically get the Angular dependency when creating a new app with npm or bower.
  - test that it works with bower: `bower install angular-esri-map#npm-bower-angular-deps`
  - test that it works with npm: `npm install https://github.com/Esri/angular-esri-map.git#npm-bower-angular-deps`

@tomwayson please review

Resolves #275

### CODE REVIEW

**Comment by jwasilgeo** [comment]
NOTE: After everything is satisfied in this PR, apply the same changes to v1 branch.

**Comment by tomwayson** [comment]
LGTM
@jwasilgeo - I like your test plan - I hadn't thought to do npm/bower installs from master like that. Go ahead and test, and then let me know how you want to handle releases for this. I assume for master you would just want to wait and bundle w/ 2.0. For 1.x, I guess 1.1.5?

@benstoltz - See test plan above, same should work for Cedar.

File: package.json [hunk 1]
```diff
@@ -3,7 +3,9 @@
   "version": "2.0.0-beta.2",
   "description": "A collection of directives to help you use Esri maps and services in your Angular applications",
   "main": "dist/angular-esri-map.js",
-  "dependencies": {},
+  "dependencies": {
+    "angular": "^1.3.0"
+  },
   "devDependencies": {
     "angular": "^1.4.7",
```
Comment (tomwayson) [inline package.json]: Don't know if conflicting dependency versions between devDependencies and dependencies will be an issue

**Comment by tomwayson** [comment]
FYI

I tested bower and npm w/ the following after merge:

```bash
bower install angular-esri-map#master
npm install https://github.com/Esri/angular-esri-map.git#master
```

In both cases the angular dependency was installed correctly. For NPM I tested both npm 2.x and 3.x. Both resolved to angular 1.5.5. Here's the npm 3 output:

```bash
npm install https://github.com/Esri/angular-esri-map.git#master
/Users/me/test/aem-install
└─┬ angular-esri-map@2.0.0-beta.2  (git+https://github.com/Esri/angular-esri-map.git#bdd67f234d68ca2a6235b00b1f788c6b90ffb706)
  └── angular@1.5.5 
```

All looks good.

**Comment by jwasilgeo** [comment]
Thanks for looking into that. For the sake of possibly cleaning up package.json anyways, is there anything you recommend we do with the angular devDependency item?

**Comment by jwasilgeo** [comment]
@tomwayson and yes, I was thinking of the same version release plan. Thanks for spelling it out to be clear (2.0 and 1.1.5). I will manually make the updates to the v1 branch today just as we did in this PR.

**Comment by tomwayson** [comment]
Re: devDependencies, I think we can just leave as is for now, but I think the right thing would be to remove angular, and set angular-mocks to ^1.3.

**Pull request merged**